src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/EvidenceTargetLinkClusterer.java

```diff
@@ -0,0 +1,122 @@
+                            // if it does, intersect the source and target intervals to refine the link
+                            it.remove();
+                            final SVInterval newSource = sourceIntervalEntry.getInterval().intersect(nextEvidence.getLocation());
+                            final SVInterval newTarget = oldLink.target.intersect(nextEvidence.getDistalTargets(readMetadata).get(0));
+                            updatedLink = new EvidenceTargetLink(newSource,
+                                    oldLink.sourceForwardStrand,
+                                    newTarget,
+                                    oldLink.targetForwardStrand,
+                                    nextEvidence instanceof BreakpointEvidence.DiscordantReadPairEvidence
+                                            ? oldLink.splitReads : oldLink.splitReads + 1,
+                                    nextEvidence instanceof BreakpointEvidence.DiscordantReadPairEvidence
+                                            ? oldLink.readPairs + 1 : oldLink.readPairs);
+                            break;
+                    }
+                }
+                if (updatedLink == null) {
+                    updatedLink = new EvidenceTargetLink(
+                            nextEvidence.getLocation(),
+                            nextEvidence.isForwardStrand(),
+                            nextEvidence.getDistalTargets(readMetadata).get(0),
+                            nextEvidence.getDistalTargetStrands(readMetadata).get(0),
+                            nextEvidence instanceof BreakpointEvidence.DiscordantReadPairEvidence
+                                    ? 0 : 1,
+                            nextEvidence instanceof BreakpointEvidence.DiscordantReadPairEvidence
+                                    ? 1 : 0);
+                }
+                currentIntervalsWithTargets.put(updatedLink.source, updatedLink);
+            }
+        }
+
+        for (Iterator<SVIntervalTree.Entry<EvidenceTargetLink>> it = currentIntervalsWithTargets.iterator(); it.hasNext(); ) {
+            links.add(it.next().getValue());
+        }
+
+        return links.iterator();
+    }
+
+    private static boolean strandsMatch(final Boolean forwardStrand1, final Boolean forwardStrand2) {
+        return forwardStrand1 != null && forwardStrand2 != null && forwardStrand1.equals(forwardStrand2);
+    }
+
+    /**
+     * Combines links that agree on interval-pair and orientation but have source and target switched. For
+     * example, if link1 has intervals s1 and t1, and link2 has s2 and t2, and s1 overlaps t1 and s2 overlaps t2,
+     * the links will be combined as long as strands agree. Returned links have the intersection of two paired intervals as
+     * source and target, with the lower-coordinate interval appearing as source.
+     */
+    public static List<EvidenceTargetLink> deduplicateTargetLinks(final List<EvidenceTargetLink> evidenceTargetLinks) {
```

Comment:
Wouldn't it be easier to always record the links in canonical order (source < target)?